Update react-button for preview release

levithomason opened the pull request:
# Update react-button for preview release

#### Pull request checklist

- [x] Add vr-test stories
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Removes all non-converged apis and concepts from react-button. Updates react-button to adhere to Fluent UI design specs.

Fixes #16820

#### Focus areas to test

Run the storybook, check examples. Consider alignment to design specs and general expected behavior.

levithomason commented on packages/make-styles/src/renderer/createDOMRenderer.ts:
```diff
@@ -58,8 +58,16 @@ export function createDOMRenderer(targetDocument: Document = document): MakeStyl
+        } catch (err) {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        }
```
This PR is also adding static styles for normalize.css, box-sizing, and font faces. This change is required to allow those to render.

layershifter replied:
This change was revert in 69597b0, correct?

fabricteam commented:
## [Perf Analysis](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ButtonNext-mount_base.html">754</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ButtonNext-mount.html">1398</a></td>
            <td>5000</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ButtonNext-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Avatar-mount_base.html">1179</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Avatar-mount.html">1248</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/BaseButton-mount_base.html">997</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/BaseButton-mount.html">1005</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Breadcrumb-mount_base.html">46098</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Breadcrumb-mount.html">45607</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ButtonNext-mount_base.html">754</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ButtonNext-mount.html">1398</a></td>
            <td>5000</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ButtonNext-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Checkbox-mount_base.html">1670</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Checkbox-mount.html">1715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/CheckboxBase-mount_base.html">1382</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/CheckboxBase-mount.html">1382</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ChoiceGroup-mount_base.html">5242</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ChoiceGroup-mount.html">5246</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ComboBox-mount_base.html">1042</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ComboBox-mount.html">1023</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/CommandBar-mount_base.html">11087</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/CommandBar-mount.html">10962</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ContextualMenu-mount_base.html">6610</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ContextualMenu-mount.html">6786</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DefaultButton-mount_base.html">1321</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DefaultButton-mount.html">1273</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DetailsRow-mount_base.html">4019</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DetailsRow-mount.html">4031</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DetailsRowFast-mount_base.html">3900</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DetailsRowFast-mount.html">3813</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DetailsRowNoStyles-mount_base.html">3754</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DetailsRowNoStyles-mount.html">3762</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Dialog-mount_base.html">1581</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Dialog-mount.html">1627</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DocumentCardTitle-mount_base.html">1879</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/DocumentCardTitle-mount.html">1883</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Dropdown-mount_base.html">3621</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Dropdown-mount.html">3515</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/FocusTrapZone-mount_base.html">1907</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/FocusTrapZone-mount.html">1916</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/FocusZone-mount_base.html">1877</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/FocusZone-mount.html">1886</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/IconButton-mount_base.html">1948</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/IconButton-mount.html">1969</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Label-mount_base.html">359</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Label-mount.html">377</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Layer-mount_base.html">1972</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Layer-mount.html">1926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Link-mount_base.html">495</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Link-mount.html">484</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/MakeStyles-mount_base.html">2111</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/MakeStyles-mount.html">2131</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/MenuButton-mount_base.html">1635</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/MenuButton-mount.html">1622</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/MessageBar-mount_base.html">2178</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/MessageBar-mount.html">2179</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Nav-mount_base.html">3646</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Nav-mount.html">3640</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/OverflowSet-mount_base.html">1106</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/OverflowSet-mount.html">1129</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Panel-mount_base.html">1549</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Panel-mount.html">1484</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Persona-mount_base.html">886</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Persona-mount.html">898</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Pivot-mount_base.html">1477</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Pivot-mount.html">1518</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/PrimaryButton-mount_base.html">1388</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/PrimaryButton-mount.html">1419</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Rating-mount_base.html">8352</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Rating-mount.html">8348</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SearchBox-mount_base.html">1465</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SearchBox-mount.html">1471</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Shimmer-mount_base.html">2839</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Shimmer-mount.html">2802</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Slider-mount_base.html">2051</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Slider-mount.html">2076</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SpinButton-mount_base.html">5380</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SpinButton-mount.html">5436</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Spinner-mount_base.html">451</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Spinner-mount.html">445</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SplitButton-mount_base.html">3353</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SplitButton-mount.html">3474</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Stack-mount_base.html">541</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Stack-mount.html">530</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/StackWithIntrinsicChildren-mount_base.html">1672</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/StackWithIntrinsicChildren-mount.html">1668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/StackWithTextChildren-mount_base.html">4891</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/StackWithTextChildren-mount.html">4971</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SwatchColorPicker-mount_base.html">11124</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/SwatchColorPicker-mount.html">11238</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Tabs</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Tabs-mount_base.html">1467</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Tabs-mount.html">1466</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/TagPicker-mount_base.html">3049</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/TagPicker-mount.html">3029</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/TeachingBubble-mount_base.html">12262</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/TeachingBubble-mount.html">12394</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Text-mount_base.html">446</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Text-mount.html">472</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/TextField-mount_base.html">1446</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/TextField-mount.html">1522</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ThemeProvider-mount_base.html">1262</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ThemeProvider-mount.html">1221</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ThemeProvider-virtual-rerender_base.html">632</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ThemeProvider-virtual-rerender.html">638</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProviderNext</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ThemeProviderNext-mount_base.html">2100</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/ThemeProviderNext-mount.html">2099</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Toggle-mount_base.html">882</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/Toggle-mount.html">907</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/buttonNative-mount_base.html">125</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test/buttonNative-mount.html">118</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

## Perf Analysis (Fluent)
<details><summary>Perf comparison</summary>

Status | Scenario | Fluent TPI | Fabric TPI | Ratio | Iterations | Ticks
:---: | :--- | ---:| ---:| ---:| ---:| ---:
🦄 | Avatar.Fluent | [0.21](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Avatar.Fluent.html) | 0.52 | 0.4:1 | 2000 | 410
🦄 | Button.Fluent | [0.14](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Button.Fluent.html) | 0.23 | 0.61:1 | 5000 | 701
🔧 | Checkbox.Fluent | [0.69](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Checkbox.Fluent.html) | 0.4 | 1.73:1 | 1000 | 688
🎯 | Dialog.Fluent | [0.17](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Dialog.Fluent.html) | 0.22 | 0.77:1 | 5000 | 867
🔧 | Dropdown.Fluent | [3.28](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Dropdown.Fluent.html) | 0.43 | 7.63:1 | 1000 | 3284
🔧 | Icon.Fluent | [0.15](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Icon.Fluent.html) | 0.06 | 2.5:1 | 5000 | 763
🦄 | Image.Fluent | [0.09](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Image.Fluent.html) | 0.14 | 0.64:1 | 5000 | 469
🔧 | Slider.Fluent | [1.77](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Slider.Fluent.html) | 0.48 | 3.69:1 | 1000 | 1771
🔧 | Text.Fluent | [0.08](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Text.Fluent.html) | 0.04 | 2:1 | 5000 | 413
🦄 | Tooltip.Fluent | [0.13](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Tooltip.Fluent.html) | 0.96 | 0.14:1 | 5000 | 636
>🔧 Needs work &nbsp; &nbsp; 🎯 On target &nbsp; &nbsp; 🦄 Amazing

</details>
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
LayoutMinimalPerf.default | [521](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/LayoutMinimalPerf.default.html) | [455](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.15:1
RefMinimalPerf.default | [263](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/RefMinimalPerf.default.html) | [238](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.11:1
AttachmentMinimalPerf.default | [229](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AttachmentMinimalPerf.default.html) | [209](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 1.1:1
ReactionMinimalPerf.default | [484](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ReactionMinimalPerf.default.html) | [450](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1.08:1
Button.Fluent | [701](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Button.Fluent.html) | [651](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Button.Fluent_base.html) | 1.08:1
ChatDuplicateMessagesPerf.default | [414](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [386](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 1.07:1
ChatMinimalPerf.default | [722](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ChatMinimalPerf.default.html) | [677](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ChatMinimalPerf.default_base.html) | 1.07:1
Avatar.Fluent | [410](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Avatar.Fluent.html) | [382](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Avatar.Fluent_base.html) | 1.07:1
BoxMinimalPerf.default | [439](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/BoxMinimalPerf.default.html) | [414](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/BoxMinimalPerf.default_base.html) | 1.06:1
ButtonUseCssPerf.default | [919](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonUseCssPerf.default.html) | [869](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonUseCssPerf.default_base.html) | 1.06:1
PortalMinimalPerf.default | [187](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/PortalMinimalPerf.default.html) | [177](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.06:1
TableMinimalPerf.default | [512](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TableMinimalPerf.default.html) | [484](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.06:1
TreeWith60ListItems.default | [220](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TreeWith60ListItems.default.html) | [207](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TreeWith60ListItems.default_base.html) | 1.06:1
FlexMinimalPerf.default | [376](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/FlexMinimalPerf.default.html) | [359](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/FlexMinimalPerf.default_base.html) | 1.05:1
GridMinimalPerf.default | [425](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/GridMinimalPerf.default.html) | [404](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.05:1
RadioGroupMinimalPerf.default | [552](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [528](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.05:1
TextAreaMinimalPerf.default | [601](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TextAreaMinimalPerf.default.html) | [575](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 1.05:1
AvatarMinimalPerf.default | [246](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AvatarMinimalPerf.default.html) | [236](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.04:1
DividerMinimalPerf.default | [479](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DividerMinimalPerf.default.html) | [461](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.04:1
ListMinimalPerf.default | [605](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListMinimalPerf.default.html) | [583](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListMinimalPerf.default_base.html) | 1.04:1
ListWith60ListItems.default | [722](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListWith60ListItems.default.html) | [696](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListWith60ListItems.default_base.html) | 1.04:1
IconMinimalPerf.default | [792](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/IconMinimalPerf.default.html) | [758](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/IconMinimalPerf.default_base.html) | 1.04:1
AnimationMinimalPerf.default | [500](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AnimationMinimalPerf.default.html) | [487](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 1.03:1
ButtonSlotsPerf.default | [685](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonSlotsPerf.default.html) | [666](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1.03:1
CheckboxMinimalPerf.default | [3053](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CheckboxMinimalPerf.default.html) | [2957](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 1.03:1
PopupMinimalPerf.default | [811](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/PopupMinimalPerf.default.html) | [790](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1.03:1
VideoMinimalPerf.default | [752](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/VideoMinimalPerf.default.html) | [730](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1.03:1
CarouselMinimalPerf.default | [553](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CarouselMinimalPerf.default.html) | [544](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 1.02:1
ListCommonPerf.default | [736](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListCommonPerf.default.html) | [722](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListCommonPerf.default_base.html) | 1.02:1
LoaderMinimalPerf.default | [795](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/LoaderMinimalPerf.default.html) | [780](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 1.02:1
ProviderMinimalPerf.default | [1124](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ProviderMinimalPerf.default.html) | [1102](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 1.02:1
StatusMinimalPerf.default | [829](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/StatusMinimalPerf.default.html) | [814](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.02:1
CardMinimalPerf.default | [652](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CardMinimalPerf.default.html) | [644](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CardMinimalPerf.default_base.html) | 1.01:1
EmbedMinimalPerf.default | [4710](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/EmbedMinimalPerf.default.html) | [4646](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 1.01:1
HeaderMinimalPerf.default | [441](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/HeaderMinimalPerf.default.html) | [435](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 1.01:1
ItemLayoutMinimalPerf.default | [1443](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [1424](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 1.01:1
ListNestedPerf.default | [683](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListNestedPerf.default.html) | [676](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ListNestedPerf.default_base.html) | 1.01:1
SkeletonMinimalPerf.default | [451](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SkeletonMinimalPerf.default.html) | [445](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1.01:1
TableManyItemsPerf.default | [2383](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TableManyItemsPerf.default.html) | [2350](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1.01:1
ToolbarMinimalPerf.default | [1125](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ToolbarMinimalPerf.default.html) | [1113](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 1.01:1
TooltipMinimalPerf.default | [943](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TooltipMinimalPerf.default.html) | [935](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1.01:1
Slider.Fluent | [1771](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Slider.Fluent.html) | [1757](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Slider.Fluent_base.html) | 1.01:1
ButtonUseCssNestingPerf.default | [1166](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonUseCssNestingPerf.default.html) | [1161](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonUseCssNestingPerf.default_base.html) | 1:1
HeaderSlotsPerf.default | [894](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/HeaderSlotsPerf.default.html) | [898](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1:1
LabelMinimalPerf.default | [485](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/LabelMinimalPerf.default.html) | [484](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/LabelMinimalPerf.default_base.html) | 1:1
SliderMinimalPerf.default | [1728](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SliderMinimalPerf.default.html) | [1731](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1:1
SplitButtonMinimalPerf.default | [4210](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [4192](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1:1
Tooltip.Fluent | [636](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Tooltip.Fluent.html) | [633](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Tooltip.Fluent_base.html) | 1:1
AlertMinimalPerf.default | [347](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AlertMinimalPerf.default.html) | [352](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AlertMinimalPerf.default_base.html) | 0.99:1
AttachmentSlotsPerf.default | [1335](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AttachmentSlotsPerf.default.html) | [1344](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 0.99:1
ButtonOverridesMissPerf.default | [1877](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1903](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 0.99:1
ImageMinimalPerf.default | [437](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ImageMinimalPerf.default.html) | [443](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ImageMinimalPerf.default_base.html) | 0.99:1
InputMinimalPerf.default | [1362](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/InputMinimalPerf.default.html) | [1370](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/InputMinimalPerf.default_base.html) | 0.99:1
ProviderMergeThemesPerf.default | [1705](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [1718](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 0.99:1
TextMinimalPerf.default | [425](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TextMinimalPerf.default.html) | [429](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TextMinimalPerf.default_base.html) | 0.99:1
CustomToolbarPrototype.default | [4138](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CustomToolbarPrototype.default.html) | [4167](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 0.99:1
TreeMinimalPerf.default | [929](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TreeMinimalPerf.default.html) | [938](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/TreeMinimalPerf.default_base.html) | 0.99:1
Dialog.Fluent | [867](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Dialog.Fluent.html) | [878](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Dialog.Fluent_base.html) | 0.99:1
DialogMinimalPerf.default | [869](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DialogMinimalPerf.default.html) | [886](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DialogMinimalPerf.default_base.html) | 0.98:1
MenuButtonMinimalPerf.default | [1780](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1817](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 0.98:1
Dropdown.Fluent | [3284](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Dropdown.Fluent.html) | [3349](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Dropdown.Fluent_base.html) | 0.98:1
Icon.Fluent | [763](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Icon.Fluent.html) | [781](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Icon.Fluent_base.html) | 0.98:1
AccordionMinimalPerf.default | [192](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AccordionMinimalPerf.default.html) | [197](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.97:1
ButtonMinimalPerf.default | [205](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonMinimalPerf.default.html) | [212](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 0.97:1
DatepickerMinimalPerf.default | [53215](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DatepickerMinimalPerf.default.html) | [54710](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 0.97:1
DropdownMinimalPerf.default | [3172](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DropdownMinimalPerf.default.html) | [3266](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 0.97:1
MenuMinimalPerf.default | [984](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/MenuMinimalPerf.default.html) | [1018](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/MenuMinimalPerf.default_base.html) | 0.97:1
RosterPerf.default | [1275](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/RosterPerf.default.html) | [1317](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/RosterPerf.default_base.html) | 0.97:1
SegmentMinimalPerf.default | [416](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SegmentMinimalPerf.default.html) | [431](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 0.97:1
Checkbox.Fluent | [688](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Checkbox.Fluent.html) | [707](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Checkbox.Fluent_base.html) | 0.97:1
Image.Fluent | [469](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Image.Fluent.html) | [483](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Image.Fluent_base.html) | 0.97:1
DropdownManyItemsPerf.default | [805](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DropdownManyItemsPerf.default.html) | [841](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 0.96:1
Text.Fluent | [413](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Text.Fluent.html) | [429](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/Text.Fluent_base.html) | 0.96:1
ChatWithPopoverPerf.default | [480](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ChatWithPopoverPerf.default.html) | [503](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 0.95:1
FormMinimalPerf.default | [503](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/FormMinimalPerf.default.html) | [528](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17060/merge/perf-test-northstar/FormMinimalPerf.default_base.html) | 0.95:1

</details><div id="perfComment9423"/>

khmakoto commented on packages/react-examples/.storybook/preview.js:
```diff
@@ -10,24 +10,28 @@ addDecorator(withPerformance);
-    'PACKAGE_NAME',
-  )
+  [
+    'react-avatar',
+    'react-badge',
+    'react-components',
+    'react-button',
+    'react-link',
+    'react-image',
+    'react-avatar',
+    'react-badge',
+    'react-link',
+    'react-image',
+    'react-menu',
```
We should remove duplicates here:
```suggestion
    'react-avatar',
    'react-badge',
    'react-button',
    'react-components',
    'react-image',
    'react-link',
```

levithomason replied:
Bad merge! Thanks for catching :D

khmakoto commented on packages/react-button/src/components/Button/Button.tsx:
```diff
@@ -12,8 +11,20 @@ import { renderButton } from './renderButton';
+  const receivedChildren = !!state.children?.children;
+  const receivedIcon = !!state.icon?.children;
+
+  const iconOnly = receivedIcon && !receivedChildren;
```
What about the `iconOnly` boolean flag specified in types?

levithomason replied:
I'm not sure that I follow this question. There is an iconOnly defined in the matcher, which is created here. Could you expound?

**A little more info**
This `iconOnly` is computed based on props received. If the user passed only an icon, then this value is true.

Since there is an `iconOnly` variant in the design spec, we pass `iconOnly` as a matcher to makeStyles. There, it is used to apply the correct styling for buttons that only have an icon.

khmakoto replied:
I meant that there is an `iconOnly` prop in the `types` file. That should either be accounted for in here or removed from the types file. That's what I meant.

khmakoto commented on packages/react-button/src/components/Button/Button.tsx:
```diff
@@ -12,8 +11,20 @@ import { renderButton } from './renderButton';
+    disabled: state.disabled,
+    primary: state.primary,
+    iconOnly: iconOnly,
+    textOnly: receivedChildren && !receivedIcon,
+    textWithIcon: receivedIcon && receivedChildren,
```
It seems like there's currently no need for these two right now looking at `useButtonClasses` so far.

levithomason replied:
Nice catch, removed. This makes me think we should have a safeguard against this somehow... I suppose the RFC on iterating styles will remove this need since we won't have matchers any longer.

khmakoto reviewed (commented):
I understand the reasoning behind trimming the `Button`. I however have a hard time justifying the removal of all the work on `MenuButton`, `SplitButton`, etc. Is there a way to maintain that in any way? Maybe keeping the files but just not exporting those from the package? I'm fine with removing stories if wanted though.

levithomason commented:
@khmakoto 
>I understand the reasoning behind trimming the Button. I however have a hard time justifying the removal of all the work on MenuButton, SplitButton, etc. Is there a way to maintain that in any way? Maybe keeping the files but just not exporting those from the package? I'm fine with removing stories if wanted though.

Yep, good call out. I tried to do this, however, they require a full overhaul from props API interfaces, styles objects, component variables, theme interface, makeStyles, makeVariants, render templates, getSlots, stories, tests, etc. Due to needing to iterate every part of them considerably it was easier to write it new than update it all.

khmakoto commented:
>> I understand the reasoning behind trimming the Button. I however have a hard time justifying the removal of all the work on MenuButton, SplitButton, etc. Is there a way to maintain that in any way? Maybe keeping the files but just not exporting those from the package? I'm fine with removing stories if wanted though.
>
> Yep, good call out. I tried to do this, however, they require a full overhaul from props API interfaces, styles objects, component variables, theme interface, makeStyles, makeVariants, render templates, getSlots, stories, tests, etc. Due to needing to iterate every part of them considerably it was easier to write it new than update it all.
>
> It is worth noting that what is in the repo right now isn't an implementation of the converged design specs and converged patterns. It was originally a v8 investigation into new patterns. Hence, there are many deps on Fabric bits and it doesn't achieve the converged needs. My understanding is that we've copied this back to v8 as well. So, the work is still there we're just removing non-converged bits from converged packages now.

@levithomason This work was not copied into v8 since we decided against promoting yet another version of the `Button` that was not converged so the work will not be anywhere. How about commenting those files completely for now? That would allow us to not remove them without having to make the abundance of changes needed for those other variants of the `Button`.

levithomason commented:
They will always be in the git history as well.

khmakoto commented:
That's true, I guess git history could serve well. Either way, I understand the reasoning for removing them now, whichever way you prefer to go I won't block based on that. I'll take a look at any new changes later today.

levithomason commented:
OK, thanks much. I'm hoping to help us remove some cruft as it has been building up and we have a lot of clean up to do.

layershifter commented on packages/react-button/src/components/Button/Button.test.tsx:
```diff
@@ -50,45 +43,32 @@ describe('Button', () => {
+  //   const testFacade = new ComponentTestFacade(Button, {});
+  //   const errors = validateBehavior(buttonBehaviorDefinition, testFacade);
+  //   expect(errors).toEqual([]);
+  // });
```
Is it expected to leave these tests commented?

layershifter replied:
_Yep, it's expected. Checked other files_

layershifter commented on packages/react-button/src/components/Button/Button.types.ts:
```diff
@@ -1,175 +1,121 @@
   ref: React.RefObject<HTMLButtonElement>;
+
+  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLSpanElement>>;
+  children?: ObjectShorthandProps<React.HTMLAttributes<HTMLSpanElement>>;
```
Are you going to keep it as it is? Or is this a temporary typing?

levithomason replied:
Temporary "fix" until we truly fix the getSlots implementation. There are TODOs added to the code about this in various places. Button should be the last component we merge for preview, we'll double back and fix hooks immediately following.

layershifter commented on packages/react-examples/.storybook/preview.js:
```diff
@@ -1,6 +1,6 @@
 // @ts-check
 import * as React from 'react';
-import { initializeIcons } from '@fluentui/font-icons-mdl2';
+import { initializeIcons } from '@fluentui/react-icons-mdl2';
```
Is it intentional or a merge conflict?

levithomason replied:
Fixed, thanks!

layershifter commented on packages/react-button/src/components/Button/useButtonClasses.ts:
```diff
@@ -1,489 +1,403 @@
+        ':global(:root)': {
+          // TODO: how to define and access component variables that do not have globals/aliases?
+          //       consider IE11 friendly way
+          '--button-height': '32px',
+          '--button-paddingX': '12px',
+          '--button-minWidth': '96px',
+          '--button-maxWidth': '280px',
+
+          '--button-iconOnly-padding': '6px',
```
I don't see any usage of these variables, cruft?

levithomason replied:
Removed, thanks. We'll address component variables once this PR is merged and we look holistically at the converged components we have.

layershifter commented on packages/react-button/src/components/Button/useButtonClasses.ts:
```diff
@@ -1,489 +1,403 @@
-      lineHeight: '1',
-    },
-  },
+        gap: buttonTokens.base.iconSpacing,
```
FYI: It's fancy property, but is not supported by the most browsers including Safari, https://caniuse.com/?search=gap.

levithomason replied:
Added TODO here for the next iteration. I want to get alpha out to early testing folks. We'll fast follow on this.

layershifter commented on packages/react-button/src/components/Button/useButtonClasses.ts:
```diff
@@ -1,489 +1,403 @@
-          contentColor: palette?.themePrimary,
-          iconColor: palette?.themePrimary,
-        },
+export const useButtonClasses = (state: ButtonState, selectors: ButtonStyleSelectors) => {
```
```suggestion
export const useButtonStyles = (state: ButtonState, selectors: ButtonStyleSelectors) => {
```

Can you please rename it to have `*Styles` suffix? It's a pattern followed in other components. And file, too 🐱

levithomason replied:
Done, thanks

khmakoto commented:
We should probably link (if not outright fix) #16820 with this PR.

levithomason commented:
> We should probably link (if not outright fix) #16820 with this PR.

@khmakoto Thanks, I fixed it in the PR description.

levithomason commented:
Addressed all feedback and added vr-tests. The rest of the TODOs I'd like to address in a fast follow but go ahead and get some early feedback on this. Looking to merge tomorrow USA time.
